lib/services/upnp-service.js

```diff
@@ -137,14 +148,24 @@ function UPnPServiceFactory(
         .then(function() {
             return _.forEach(self.registry, function(r) {
                 logger.debug('Starting SSDP/uPnP server', {usn: r});
-                self.options.location = (endpoint.httpsEnabled ? 'https':'http') + 
-                    '://' + (endpoint.address !== '0.0.0.0' ? endpoint.address: ip.address()) + 
-                    ':' + endpoint.port + r.path;
-                var server = new Server(self.options);
+                var ep = (_.includes(r.urn, 'southbound')) ? southPoint : endpoint;
+                var loptions = {
+                    location: (ep.httpsEnabled ? 'https':'http') + 
+                        '://' + (ep.address !== '0.0.0.0' ? ep.address: ip.address()) + 
+                        ':' + ep.port + r.path
+                }
+                var server = new Server(_.merge({}, self.options, loptions));
                 server.addUSN(r.urn);
                 server.on('advertise-alive', _alive.bind(self)); //jshint ignore: line
                 server.on('advertise-bye', _bye.bind(self));    //jshint ignore: line
-                server.start(self.bindAddress);
+                server.start();
+                server.sock.on('listening', function() {
+                    try {
+                        server.sock.addMembership(server._ssdpIp, ep.address);
+                    } catch(e) {
+                        logger.debug('addMembership', e);
```

Comment:
The logger is already tagged with the Upnp service name from the providers export:
[debug] [2016-05-27T19:03:48.745Z] [on-http] [Http.Services.uPnP] [Server] addMembership
 -> /lib/services/upnp-service.js:166